### PR TITLE
Add Stacked Avatar

### LINKED ISSUE #709: Feature: Stacked avatar for issue participants
### Describe the solution you'd like

Feature: Stacked avatar for issue participants

### Describe alternatives you've considered

GitHub Primer Stacked Avatar

### Additional context

POC: https://codepen.io/ygcodes/pen/pogNmVJ

Screenshot

### PR DESCRIPTION
**Pull request by yg**
## What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [x] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

## Description

#709 

## Related Tickets & Documents

Closes #709 

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)

https://www.loom.com/share/e7f27904153c44a69c29f12e513eb948

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 readme
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/components/IssueListItem.js [hunk 1]
```diff
@@ -41,9 +41,11 @@ function IssueListItem({title, labels, author, opened, type, participants, comme
                   {milestone.title}
                 </span>
               )}
-              {participants && participants.nodes.map((user, key) => (
-                <img className="participants" key={key} src={user.avatarUrl} title={user.login} />
-              ))}
+              <div className="avatar-stack">
+                {participants && participants.nodes.map((user, key) => (
+                  <img className="avatar" key={key} src={user.avatarUrl} title={user.login} />
```
Comment (sujitagarwal) [inline src/components/IssueListItem.js:46]: a valid URL format may contain space character (bad practice, but happens) as well.. I'd suggest adding the value for src within quotes.

File: src/styles/List.js [hunk 2]
```diff
@@ -20,13 +20,28 @@ const Container = styled.ul`
...
-    margin-right: 0;
-    margin-left: 5px;
-    margin-top: 5px;
+    display: inline;
+    margin-left: 3px;
+
+    .avatar {
+      height: ${size.tiny};
+      margin-right: -${size.micro};
+      margin-top: 10px;
```
Comment (sujitagarwal) [inline src/styles/List.js:31]: I think maintaining a consistent approach to feed values either in static/dynamic form should be followed. For margin-right it comes from a variable. For margin-top it is fed static.

File: src/styles/List.js [hunk 3]
```diff
@@ -20,13 +20,28 @@ const Container = styled.ul`
...
+      background-color: white;
+      border: 1px solid white;
```
Comment (sujitagarwal) [inline src/styles/List.js:33]: If the background color is already set to white, do we need to put border as well? I think the border is not needed.

File: src/components/IssueListItem.js (hunk 1 above)
Comment (sujitagarwal) [inline src/components/IssueListItem.js:46]: if AvatalURL is empty, a placeholder image can be used.

Comment (yg) [reply src/components/IssueListItem.js:46]: @sujitagarwal In browser it renders the URL in quotes

```html
<img class="avatar" src="https://avatars3.githubusercontent.com/u/67512?v=4" title="dougwilson">
```

Comment (sujitagarwal) [reply src/components/IssueListItem.js:46]: That's a browser specific feature, and quite unreliable when trying out with some unpopular ones

File: src/styles/List.js (hunk 3 above)
Comment (yg) [reply src/styles/List.js:33]: @sujitagarwal The images seem merged without a border!

| Without Border | With Border |
|---|---|
| Screenshot | Screenshot |

Comment (bdougie) [reply src/styles/List.js:33]: with-border matches what GitHub does as well.

Comment (sujitagarwal) [reply src/styles/List.js:33]: I see... In this scenario it is fine then...

File: src/components/IssueListItem.js (hunk 1 above)
Comment (yg) [reply src/components/IssueListItem.js:46]: @sujitagarwal If `avatarUrl` is empty will not render the image at all!

Comment (sujitagarwal) [reply src/components/IssueListItem.js:46]: Could you please explain me the use-case? I'm trying to figure out why we're having this behavior.

Also, what's the upper limit on number of users being shown?

Comment (yg) [reply src/components/IssueListItem.js:46]: @sujitagarwal React renders the same across all the browsers!

http://blog.matthewcheok.com/a-better-img-tag/

**Comment by filiptronicek** [comment]
@yg what about bumping the max participants count then?

File: src/components/IssueListItem.js (hunk 1 above)
Comment (sujitagarwal) [reply src/components/IssueListItem.js:46]: Great.

File: src/styles/List.js [hunk 4]
```diff
@@ -20,13 +20,28 @@ const Container = styled.ul`
...
+      border-radius: ${size.tiny};
+      transition: margin 0.1s ease-in-out;
+
+      &:last-child {
+        margin-right: 2px;
```
Comment (yg) [inline src/styles/List.js:38]: > @yg what about bumping the max participants count then?

@filiptronicek We already bumping it here!

**Comment by Brawrdon** [comment]
I think now with all the information added to that "meta data" line the ordering of the elements is getting a little confusing, specifically having the tags in the middle is a little distracting. This isn't directly related to this PR so I think another issue should be made for it. I think something like this (with the padding fixed) would be easier to digest!

image

**Comment by JapneetSingh5** [comment]
> @yg what about bumping the max participants count then?

I think we can have a max limit, say 5, and then if we have, say 10 users, we can show 4 avatars with a 5th circle having a +6

**Comment by JapneetSingh5** [comment]
> > @yg what about bumping the max participants count then?
> 
> I think we can have a max limit, say 5, and then if we have, say 10 users, we can show 4 avatars with a 5th circle having a +6

Screen Shot 2020-06-18 at 11 38 39 PM
Discord has a similar implementation for minimised voice calls

**Comment by yg** [comment]
> I think we can have a max limit, say 5, and then if we have, say 10 users, we can show 4 avatars with a 5th circle having a +6

Good suggestion @JapneetSingh5! I need some help doing that!

If anyone can do it, dive into the Codepen https://codepen.io/ygcodes/pen/pogNmVJ and help me with the changes!

**Comment by filiptronicek** [comment]
What about this @yg?
suggestion with the remaining avatars in a small circle on the bottom

**Comment by yg** [comment]
> What about this @yg?
> suggestion with the remaining avatars in a small circle on the bottom

@filiptronicek It will be not visible in open sauced, cuz we are using very small avatars!

**Comment by filiptronicek** [comment]
Oh... right, didn't think of that. My bad. I will try to make a better suggestion

**Comment by JapneetSingh5** [comment]
Edited the pen, added some basic hover animation too.
https://codepen.io/japneetsingh5/pen/OJMWQKG
@yg 
Screen Shot 2020-06-19 at 7 36 56 PM

**Comment by yg** [comment]
> Edited the pen, added some basic hover animation too.

Thanks @JapneetSingh5, Can you please save it on Codepen! I'm still seeing the old version!

**Comment by JapneetSingh5** [comment]
> > Edited the pen, added some basic hover animation too.
> 
> 
> 
> Thanks @JapneetSingh5, Can you please save it on Codepen! I'm still seeing the old version!

Saved it, should show the right one now.

**Comment by yg** [comment]
Screenshot

Thanks, @JapneetSingh5! :heart:

**Comment by JapneetSingh5** [comment]
> Screenshot
> 
> 
> 
> Thanks, @JapneetSingh5! :heart: 

Amazing 😍

File: src/styles/List.js (hunk 2 above)
Comment (bdougie) [reply src/styles/List.js:31]: Agreed, this should be:

```js
margin-top: ${margin.gutter}
```

Comment (yg) [reply src/styles/List.js:31]: Done :+1:

**Pull request merged**